clinica/pydra/engine_utils.py

```diff
@@ -49,18 +49,27 @@ def bids_query(keys: list) -> dict:
 
 
 def run(wf: Workflow) -> None:
-
-    """Execute a Pydra workflow
-
-    Parameters
-    ----------
-        wf : Workflow
-            The workflow to execute
-    """
-
-    with Submitter(plugin="cf") as submitter:
-        submitter(wf)
-
-    results = wf.result(return_inputs=True)
-    # @TODO: decide where to store results
-    print(results)
+    import re
+
+    try:
+        with Submitter(plugin="cf") as submitter:
+            submitter(wf)
+    except Exception as e:
+        path = re.search("\/.*\.pklz", str(e))
+        if path:
+            path = path.group(0)
+            read_error(path)
+        else:
+            print("Exception:", e)
+        return str(e)
+
+    results = wf.result(return_inputs=False)
+    return results
+
+
+def read_error(path):
+    import cloudpickle as cp
+
+    with open(path, "rb") as fp:
+        err = cp.load(fp)
+        print(err["error message"])
```

Comment:
I would either return the error, or change the name of this function to `print_error`. WDYT?